src/codeu/chat/client/core/View.java

```diff
@@ -45,6 +45,23 @@ public View(ConnectionSource source) {
     this.source = source;
   }
 
+  public ServerInfo getInfo() {
+    try (final Connection connection = this.source.connect()) {
+      Serializers.INTEGER.write(connection.out(), NetworkCode.SERVER_INFO_REQUEST);
+      if (Serializers.INTEGER.read(connection.in()) == NetworkCode.SERVER_INFO_RESPONSE) {
+        final Uuid version = Uuid.SERIALIZER.read(connection.in());
+        return new ServerInfo(version);
+      } else {
+        LOG.error("Unexpected Server Response.");
+      }
+    } catch (Exception ex) {
+      System.out.println("ERROR: Exception while connecting. Check log for details.");
```

Comment:
i see that this pattern has been used in the skeleton code, so you can ignore this comment for now.